File: ExtendedSchedule/domain/aggregate/user_config.go
package aggregate

import (
	"tellmeac/extended-schedule/domain/entity"

	"github.com/google/uuid"
)

func NewUserConfig(email string) UserConfig {
	return UserConfig{
		ID:    uuid.New(),
		Email: email,
	}
}

// UserConfig представляет объект конфигурации пользователя.
type UserConfig struct {
	ID                   uuid.UUID               `json:"id"`
	Email                string                  `json:"email"`
	BaseGroup            *entity.GroupInfo       `json:"baseGroup"`
	ExcludedLessons      []entity.ExcludedLesson `json:"excludedLessons"`
	ExtendedGroupLessons []ExtendedGroupLessons  `json:"extendedGroupLessons"`
}

// ExtendedGroupLessons представляет группу с дополнительными предметами.
type ExtendedGroupLessons struct {
	Group     entity.GroupInfo `json:"group"`
	LessonIDs []string         `json:"lessonIds"`
}

File: web/src/SettingsPage/Models/UserConfig.ts
import {GroupInfo} from "../../Shared/Models";

/**
 * User configuration
 */
export interface UserConfig {
    id: string
    email: string
    baseGroup: GroupInfo
    excludedLessons: ExcludedRule[]
    extendedGroupLessons: ExtendedLessons[]
}

/**
 * Exclude rule for lessons
 * TODO: feature
 */
export interface ExcludedRule {}

/**
 * ExtendedLessons is set of lessons to include from group to user's schedule
 */
export interface ExtendedLessons {
    group: GroupInfo
    lessonIds: string[]
}

/**
 * LessonInfo without schedule day context
 */
export interface LessonInfo {
    id: string
    name: string
    teacherName: string
    lessonKind: string
}